### PR TITLE
Fix bug with no props being passed down in 2.2.0, closes #1849

### LINKED ISSUE #1849: RefinementList createURL breaks in 2.2.0
Browsing the changelog myself, if I see the problem I'll comment here with an update. 

Edit: Looks like onSearchStateChanged is never being called either ?

**Bug: What is the current behavior?**
Calling createURL on a refinement just returns `#` in 2.2.0, revert back to 2.1.0, works fine.

**Bug: What is the expected behavior?**
For it to work :) 

**What project are you opening an issue for?**
- react-instantsearch

Edit 2, might want some tests that check for injected variables from createInstanteSearch? :)

### PR DESCRIPTION
**Pull request by zackify**
#1849

### CODE REVIEW

**Comment by bobylito** [comment]
Thanks for the contribution @zackify :)

File: packages/react-instantsearch/src/core/createInstantSearch.js [hunk 1]
```diff
@@ -41,6 +41,7 @@ export default function createInstantSearch(defaultAlgoliaClient, root) {
     render() {
       return (
         <InstantSearch
+          {...this.props}
```
Comment (bobylito) [inline packages/react-instantsearch/src/core/createInstantSearch.js]: I don't think we want to go back to this spread way of doing things. Best would be to be explicit about all the parameters that will be passed down. It seems that we're missing two things: 
 - `searchParameters`
 - `createURL`

Comment (mthuret) [inline packages/react-instantsearch/src/core/createInstantSearch.js]: We also need:

* searchState
* onSearchStateChange

Comment (bobylito) [inline packages/react-instantsearch/src/core/createInstantSearch.js]: Did that in another PR https://github.com/algolia/instantsearch.js/pull/1867

**Pull request closed without merging**